diambra/arena/arena_gym.py

```diff
@@ -130,411 +99,244 @@ def render(self, mode='human', wait_key=1):
+        observation_space_dict.update(generic_obs_dict)
+        observation_space_dict["agent_0"] = gym.spaces.Dict(player_obs_dict)
+        observation_space_dict["agent_1"] = gym.spaces.Dict(player_obs_dict)
 
-            if k == "stage":
-                continue
+        self.observation_space = gym.spaces.Dict(observation_space_dict)
 
-            if k[-2:] == "P1":
-                knew = "own" + k[:-2]
-            else:
-                knew = "opp" + k[:-2]
+        # Action space
+        # Dictionary
+        action_spaces_values = {"multi_discrete": gym.spaces.MultiDiscrete(self.n_actions),
+                                "discrete": gym.spaces.Discrete(self.n_actions[0] + self.n_actions[1] - 1)}
+        action_space_dict = self._update_dict(action_spaces_values)
+        self.logger.debug("Using the following action spaces: {}".format(action_space_dict))
+        self.action_space = gym.spaces.Dict(action_space_dict)
 
-            if v[0] == 0 or v[0] == 2:  # Discrete spaces
-                player_spec_dict[knew] = spaces.Discrete(v[2] + 1)
-            elif v[0] == 1:  # Box spaces
-                player_spec_dict[knew] = spaces.Box(low=v[1], high=v[2],
-                                                    shape=(1,), dtype=np.int32)
+    # Return the no-op action
+    def get_no_op_action(self):
+        no_op_values = {"multi_discrete": [0, 0],
+                        "discrete": 0}
+        return self._update_dict(no_op_values)
 
+    # Step the environment
+    def step(self, actions: Dict[str, Union[int, List[int]]]):
+        # NOTE: the assumption in current interface is that we have actions sorted as agent's order
+        actions = sorted(actions.items())
+        for idx, action in enumerate(actions):
+            # Defining move and attack actions P1/P2 as a function of action_space
+            if isinstance(self.action_space[action[0]], gym.spaces.MultiDiscrete):
+                self._last_action[idx] = action[1]
             else:
-                raise RuntimeError("Only Discrete and Box Spaces allowed")
-
-        actions_dict = {
-            "move": spaces.Discrete(self.n_actions[0][0]),
-            "attack": spaces.Discrete(self.n_actions[0][1])
-        }
+                self._last_action[idx] = list(discrete_to_multi_discrete_action(action[1], self.n_actions[0]))
+        response = self.arena_engine.step(self._last_action)
+        observation = self._get_obs(response)
 
-        player_spec_dict["actions"] = spaces.Dict(actions_dict)
-        player_dict_p1 = player_spec_dict.copy()
-        observation_space_dict["P1"] = spaces.Dict(player_dict_p1)
+        return observation, response.reward, response.info.game_states["game_done"], False, self._get_info(response)
 
-        actions_dict = {
-            "move": spaces.Discrete(self.n_actions[1][0]),
-            "attack": spaces.Discrete(self.n_actions[1][1])
-        }
+    def _update_dict(self, values_dict):
```

Comment:
Can we come up witth a better name? Chatgpt suggests map_action_space_to_agents :)

Reply:
Sure